src/config/options.rs

```diff
@@ -141,6 +141,25 @@ configuration_option_enum! { IndentStyle:
     Block,
 }
 
+configuration_option_enum! { MatchReferenceStyle:
+    Manual,
+    // match x {
+    //    &A => ...,
+    //    ...
+    // }
+    Reference,
+    // match *x {
+    //     A => ...,
+    //     ...
+    // }
+    Dereference,
+    // match x {
+    //     A => ...,
```

Comment:
I think this should `&A`

Reply:
Rust 1.26 has [stabilized match_default_bindings](https://github.com/rust-lang/rust/pull/49394)

>the compiler automatically references the Some, and since we’re borrowing, name is bound as ref name automatically as well. If we were mutating:
```rust
fn hello(arg: &mut Option<String>) {
    match arg {
        Some(name) => name.push_str(", world"),
        None => (),
    }
}
```
>the compiler will automatically borrow by mutable reference, and name will be bound as ref mut too.

Reply:
Hmm, then should it be `*x`? In any case the comments here don't seem to match the examples in the PR message. Perhaps it would be good to explain the reference/dereference modes here.

What is the motivation for having four modes? I would have thought that auto and manual would be the most widely used.

Reply:
Before Rust 1.26, you can choose `reference`/`dereference` modes as you wish, just different code style in same semantic in most cases.

> the compiler automatically references the Some, and since we’re borrowing, name is bound as ref name automatically as well.

Since Rust 1.26, the compiler could do smarter, it can automatic handle the `reference`/`dereference` modes base on the type inferencing, that's why we need a new `auto` mode.

```rust
fn hello(name: &Option<&str>) {
    match name {
        Some(name) => println!("Hello {}!", name),
        None => println!("I don't know who you are."),
    }
}

fn main() {
    let name = Some("world");

    hello(&name);
}
```

So, if we use Rust 1.26 or later, just set the `auto` mode and migrate the old code, let's compiler do the dirty works.

Reply:
Right, but since 1.26 is stable now, I think we could just offer auto and manual, and we don't need to offer ref and deref styles. Do you think there is an advantage to continue to offer ref and deref?

Reply:
The problem is some projects may still using the old Rust version, they may need to align the code style with `rustfmt` for ref/deref styles.